csvy/readers.py

```diff
@@ -270,3 +271,52 @@ def read_to_list(
             data.append(row)
 
     return data, header
+
+
+def read_into_dict(file_path: Path | str):
```

Comment:
This should have the same input parameters as the other read functions to maintain a consistent API.